src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/InputRightCurlyBracePolicyAloneForAnonymousClass.java

```diff
@@ -0,0 +1,107 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+class InputRightCurlyBracePolicyAloneForAnonymousClass {
+
+    private static final Set<String> TREE_SET = Collections.unmodifiableSortedSet(
+            Arrays.stream(new String[] {"br", "li", "dt", "dd", "hr", "img", "p", "td", "tr", "th", })
+                    .collect(Collectors.toCollection(TreeSet::new))); // for coverage
+    public static final String STRING = new String("String");
+    public static final Object OBJECT1 = new Object();
+    public static final Object OBJECT2 = new Object() {
+        @Override
+        public String toString() {
+            return "";
+        }
+    }; int field1;   // violation
+
+    int field2;
+
+
+    static void method1(Object o) {
+        new Object() { @Override public String toString() { return ""; } int binary = 10; }; // violation
+    }
+
+    static void method2(Thread t, int x) {
+        method1(new Object() {
+            @Override
+            protected Object clone() {
+                return new Object() { @Override protected void finalize() { "".toString(); } }; // violation
+            }
+        }); // no violation
+    }
+
+    void method3() {
+        method2(new Thread() {
+            @Override
+            public void run() {
+                start();
+                super.run();
+            }
+        }, field1); // violation
+
+
+
+        method2(new Thread() {
+                    @Override
+                    public void run() {
+                        field1++;
+                        start();
+                        super.run();
+                    }
+                },      // no violation
+                0);
+    }
+
+    static Thread method4(Runnable r, String s) {
+        r.run();
+        return new Thread(r,s) {
+            @Override
+            public String toString() {
+                return "name:"+s;
+            }
+        }; // no violation
+    }
+
+    static Thread method4(Runnable r) {
+        r.run();
+        String s = new Object().toString();
+        return new Thread(r, new Object() {
+            @Override
+            public String toString() {
+                return s;
+            }
+        }.toString());  // violation
+    }
+
+    void method5(int t) {
+        method2(method4(new Runnable() {
+                    @Override
+                    public void run() {
+                        field1+=field2;
+                        field2++;
+                        if (t>field1) {
+                            run();
+                        }
+                    }
+                }),        // no violation
```

Comment:
@romani @rnveach 
From my point of view this is not OK. There should be a violation.

Reply:
@romani @MEZk @rnveach 
This Point was left out. Regression reports for option _alone_ and _alone-or-singleline_ have been approved. There is still no violation here. Please confirm if a violation is needed here.

Reply:
As for me, violation should be there for `alone` and `alone-or-singleline`, but it should not be there at `same` http://checkstyle.sourceforge.net/property_types.html#rcurly
>
```
    Thread t = new Thread(new Runnable() {

    }); // this is OK, allowed for better code readability
```

Reply:
@romani 
Done for `same` and `alone`. As far as `alone_or_singleline` goes, [`isAnonInnerClassInit`](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java#L268) prevents an error. This method was added [in this commit](https://github.com/checkstyle/checkstyle/commit/f1dced161ab0c818f0e1e125fc5b88b71f5b171e)

Reply:
> As far as alone_or_singleline goes, isAnonInnerClassInit prevents an error.

make sense to have no error.